app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  before_create :api_key_creation
+  has_secure_password
+
+  validates :email, presence: true, uniqueness: true
+  validates :password, presence: true, confirmation: {allow_blank: false }
+
+private
+  def api_key_creation
+    self.api_key ||= SecureRandom.hex(10) 
```

Comment:
This felt cool. I was getting a random key every time and managed to throw in the || without really thinking and it worked the way I needed it to!